frontend/js/marc-record-merge-postmerge-service.js

```diff
@@ -252,6 +252,21 @@ export function removeExtra035aFromMerged(preferredRecord, otherRecord, mergedRe
   };
 }
 
+export function mergeUniqueF042(preferredRecord, otherRecord, mergedRecordParam) {
+  const mergedRecord = new MarcRecord(mergedRecordParam);
+  const f042s = mergedRecord.get(/^042$/);
+  if (f042s.length > 0) {
+    const uniqueValues = [...new Set([].concat(...f042s.map(field => field.subfields.map(sub => sub.value))))];
+    const uniqueSubs = uniqueValues.map(value => {return {code: 'a', value}})
+    mergedRecord.fields = mergedRecord.fields.filter(field => field.tag !== '042');
```

Comment:
Set won't help you here...The equality comparison is is not deep. You need to use reduce. Also, the spec doesn't specify that only `a` subfields should be retained.
```
const subfields = f042s
.map(({subfields}) => subfields)
.reduce((acc, ({code, value})) => 
  return acc.some(isEqual) ? acc : acc.concat({code, value});

  // Check if subfields have equal code and value
  function isEqual() {
  }
, []);
```

Reply:
https://marc21.kansalliskirjasto.fi/bib/01X-04X.htm#042 042 should have only a